### PR TITLE
CRAFTERCMS-3363

### PR DESCRIPTION
**Pull request by dejan-brkic**
Added write global configuration feature

### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/3363

### CODE REVIEW

File: src/main/resources/crafter/studio/upgrade/pipelines.yaml [hunk 1]
```diff
@@ -560,6 +560,23 @@ pipelines:
...
+        - type: xsltFileUpgrader
+          path: /configuration/global-menu-config.xml
+          template: crafter/studio/upgrade/3.1.x/3.1.4.9/system/global-menu-config.xslt
+          commitDetails: Added global configuration menu item
+        - type: globalRepoUpgrader
+          files:
+            - src: crafter/studio/upgrade/3.1.x/3.1.4.9/system/sample-studio-config-override.yaml
+              dest: /configuration/samples/sample-studio-config-override.yaml
+          commitDetails: Update sample files
+        - type: dbVersionUpgrader
```
Comment (jdrossl) [inline src/main/resources/crafter/studio/upgrade/pipelines.yaml:579]: Please update the version in the SQL script too

File: src/main/java/org/craftercms/studio/api/v2/service/config/ConfigurationService.java [hunk 2]
```diff
@@ -105,4 +110,6 @@ Resource getPluginFile(String siteId, String type, String name, String filename)
      * @return configuration history
      */
     ConfigurationHistory getConfigurationHistory(String siteId, String module, String path, String environment);
+
+    void writeGlobalConfiguration(String path, InputStream content) throws ServiceLayerException;
```
Comment (jdrossl) [inline src/main/java/org/craftercms/studio/api/v2/service/config/ConfigurationService.java:124]: Shouldn't this also need the `write` permission?

**Review by jdrossl — CHANGES_REQUESTED** [review]

File: src/main/java/org/craftercms/studio/api/v2/service/config/ConfigurationService.java (hunk 2 above)
Comment (dejan-brkic) [reply src/main/java/org/craftercms/studio/api/v2/service/config/ConfigurationService.java:124]: Interfaces do not need permission

File: src/main/java/org/craftercms/studio/api/v2/service/config/ConfigurationService.java [hunk 3]
```diff
@@ -16,6 +16,8 @@
  */
 package org.craftercms.studio.api.v2.service.config;
 
+import org.craftercms.commons.security.permissions.DefaultPermission;
```
Comment (jdrossl) [inline src/main/java/org/craftercms/studio/api/v2/service/config/ConfigurationService.java]: Please remove unused imports

File: src/main/java/org/craftercms/studio/impl/v2/service/configuration/ConfigurationServiceImpl.java [hunk 4]
```diff
@@ -188,6 +190,12 @@ public Document getConfigurationAsDocument(@ProtectedResourceId(SITE_ID_RESOURCE
         return retDocument;
     }
 
+    @Override
+    @HasPermission(type = DefaultPermission.class, action = "write_global_configuration")
```
Comment (jdrossl) [inline src/main/java/org/craftercms/studio/impl/v2/service/configuration/ConfigurationServiceImpl.java:194]: Same comment as before, shouldn't we add both `read` & `write` permission?

**Review by jdrossl — CHANGES_REQUESTED** [review]

File: src/main/java/org/craftercms/studio/impl/v2/service/configuration/ConfigurationServiceImpl.java (hunk 4 above)
Comment (sumerjabri) [reply src/main/java/org/craftercms/studio/impl/v2/service/configuration/ConfigurationServiceImpl.java:194]: We'll proceed with `write` implies `read` and for this tool, you need `write` to do anything meaningful.

**Review by jdrossl — APPROVED** [review]

**Review by sumerjabri — APPROVED** [review]

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**